### PR TITLE
[16.0][MIG] base_phone: Migration to 16.0

### PR DESCRIPTION
**Pull request by nguyenminhchien**
This PR replaces https://github.com/OCA/connector-telephony/pull/294

### CODE REVIEW

**Review by leemannd — APPROVED** [review]

**Comment by StephaneMangin** [comment]
Please anonymise the test datas

**Comment by mmequignon** [comment]
<s>Hello, I commented on the previous pr:
https://github.com/OCA/connector-telephony/pull/294#discussion_r1123058362

This mixin looks useless to me, since `odoo.addons.phone_validation.tools.phone_format` has been introduced.
Therefore, this whole module would be useless as well.
I think we should use it in https://github.com/OCA/connector-telephony/pull/297 and drop this `base_phone` module if possible.</s>

**Comment by nguyenminhchien** [comment]
> Please anonymise the test datas

@StephaneMangin : updated, please review.

**Review by mmequignon — APPROVED** [review]

File: base_phone/tests/test_base_phone.py [hunk 1]
```diff
@@ -61,14 +61,14 @@ def test_partner_phone_formatting(self):
...
                 "is_company": True,
             }
         )
         partner2 = rpo.create(
             {
-                "name": "Joël Grand-Guillaume",
+                "name": "Partner 2",
                 "parent_id": parent_partner2.id,
                 "phone": "(0) 21 619 10 10",
                 "mobile": "(0) 79 606 42 42",
```
Comment (StephaneMangin) [inline base_phone/tests/test_base_phone.py]: These numbers are camptocamp ones. Please fully anonymise all the data.

**Review by StephaneMangin — CHANGES_REQUESTED** [review]

File: base_phone/tests/test_base_phone.py (hunk 1 above)
Comment (StephaneMangin) [reply base_phone/tests/test_base_phone.py]: cc @nguyenminhchien

Comment (hailangvn2023) [reply base_phone/tests/test_base_phone.py]: Good day @StephaneMangin, [new PR](https://github.com/OCA/connector-telephony/pull/300) has been created to replace this PR and to update the test data.

**Pull request closed without merging**